File: c/src/Behavioral/Visitor/DvdInfo.c


//DvdInfo - one of three concrete Visitees

#include "TitleInfo.h"
#include "DvdInfo.h"

#include "stdlib.h"
#include "mem.h"
#include "assert.h"

void DvdInfo_accept( void *  this, TitleBlurbVisitor_t * tbv) 
{
	tbv->visit(tbv,this);
}

    
TitleInfo_t * DvdInfo_new(char * titleName, char * star, char encodingRegion) 
{
	TitleInfo_t * bi = TitleInfo_new( DvdInfo_accept, titleName );
	bi->type = DvdInfoType;
	NEW(bi->dvd);

	bi->dvd->star = star;
	bi->dvd->encodingRegion = encodingRegion;
	return bi;
}    

void DvdInfo_free(TitleInfo_t * d)
{
	assert(d );
	FREE( d->dvd );
	FREE(d);	
}

   

File: go/behavioral/visitor/DvdInfo.go
package visitor

import (
	"fmt"
)

type DvdInfo struct {  
	Title
   star , encodingRegion string
}
    
func NewDvdInfo(title string , star string, encodingRegion string) *DvdInfo {
	return &DvdInfo{ &TitleInfo{title}, star, encodingRegion}
}    
   
func (this *DvdInfo) Accept( tbv Visitor) {
	tbv.Visit(this)
}

func (this *DvdInfo) GetLongBlurb() string { 
	return fmt.Sprintf("LB-DVD: %s, starring %s, encoding region %s",
		this.GetTitle() , 
		this.star  , 
		this.encodingRegion)
}


func (this *DvdInfo) GetShortBlurb() string { 
	return fmt.Sprintf("SB-DVD: %s",
		this.GetTitle() )
}

